lib/concurrent-ruby/concurrent/configuration.rb

```diff
@@ -1,7 +1,14 @@
 require 'thread'
 require 'concurrent/delay'
 require 'concurrent/errors'
-require 'concurrent/atomic/atomic_reference'
+
+if Concurrent.on_truffleruby?
+  # Avoid TruffleRuby's patch
+  require File.expand_path('atomic/atomic_reference', __dir__)
+else
+  require 'concurrent/atomic/atomic_reference'
+end
```

Comment:
This workaround is unnecessary if we use truffleruby-head or can wait the next release

Reply:
It's https://github.com/oracle/truffleruby/issues/2605 BTW

Reply:
What is the actual issue with the patch? I couldn't find it out.

Reply:
I more mean - why do we need the patch anymore?

Reply:
TruffleRuby needs the patch for older versions of concurrent-ruby, which seems important they still work.
The patch should have no effect whatsoever on newer concurrent-ruby versions, so it's fine to "always apply it".

This workaround OTOH feels hacky and might not work reliably so I don't want to merge that.